Update dotnet-core-sdk for windows and darwin

grospelliergilles opened the pull request:
<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

grospelliergilles committed:
dotnet-core-sdk: add version 8.0.21 binaries for 'darwin' and 'windows' platforms

tldahlgren commented on repos/spack_repo/builtin/packages/dotnet_core_sdk/package.py:
```diff
@@ -41,6 +42,34 @@ class DotnetCoreSdk(Package):
             url="https://builds.dotnet.microsoft.com/dotnet/Sdk/6.0.428/dotnet-sdk-6.0.428-linux-arm64.tar.gz",
             sha256="b5956b0d9ab3063c2886ec74adc953394e81a1aa3f5075c6b41b3e4f1d7a53f9",
         )
+    elif platform.system() == "darwin" and platform.machine() == "x86_64":
```
The first letter returned by `platform.system()` is in upper case.

You can see this by running:
```
$ spack python
Spack version 1.1.0.dev0
Python 3.13.5, Darwin arm64
>>> import platform
>>> platform.system()
'Darwin'
>>> exit()
```

tldahlgren replied:
I'll have to trust you on the sha256 because I'm not getting the value with this fix or when running `sha256sum < <archive-file>` on/for the above machine.

tldahlgren reviewed (changes requested):
See comment.

grospelliergilles commented:
Thanks @tldahlgren for the quick review. I have fixed the uppercase for platform.
You were right, some hash were bad because my downloads were truncated. I have also fixed that.

tldahlgren reviewed (approved):
Thanks.
